### PR TITLE
feat: implement Gitea alerting provider integration

### PR DESCRIPTION
**Pull request by appleboy**
## Summary

- Add TypeGitea for the gitea alerting provider
- Introduce a new file for the gitea alerting provider implementation
- Implement the AlertProvider struct with necessary fields for gitea integration
- Add validation logic for the AlertProvider configuration
- Create tests for the AlertProvider's validation and sending functionality
- Update go.mod to include the gitea SDK as a dependency
- Modify the alerting configuration validation to recognize TypeGitea
- Add Assignees support to AlertProvider functionality

gitea-alerts

## Checklist

- [X] Tested and/or added tests to validate that the changes work as intended, if applicable.
- [X] Updated documentation in `README.md`, if applicable.

### CODE REVIEW

**Comment by appleboy** [comment]
@TwiN The PR is already done. Please help to review.

File: alerting/provider/gitea/gitea.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+// AlertProvider is the configuration necessary for sending an alert using Discord
+type AlertProvider struct {
+	RepositoryURL string `yaml:"repository-url"` // The URL of the Gitea repository to create issues in
+	Token         string `yaml:"token"`          // Token requires at least RW on issues and RO on metadata
+
+	// DefaultAlert is the default alert configuration to use for endpoints with an alert of the appropriate type
+	DefaultAlert *alert.Alert `yaml:"default-alert,omitempty"`
+
+	// SkipVerify disables SSL certificate verification
+	SkipVerify bool `yaml:"skip-verify,omitempty"`
```
Comment (TwiN) [inline alerting/provider/gitea/gitea.go:24]: For consistency, this should be replaced by `client.Config`, which would enable `alerting.gitea.client.insecure` to achieve the same goal.

See https://github.com/TwiN/gatus/blob/dd435a8eafa604b0cb5ed9355adb17fdca48aba7/alerting/provider/googlechat/googlechat.go#L20 for an example of another provider that uses it.

File: README.md [hunk 2]
```diff
@@ -697,6 +699,43 @@ endpoints:
...
+      - "[RESPONSE_TIME] < 75"
+    alerts:
+      - type: gitea
+        failure-threshold: 2
+        success-threshold: 3
+        send-on-resolved: true
+        description: "Everything's burning AAAAAHHHHHHHHHHHHHHH"
+```
+
+![GitHub alert](.github/assets/gitea-alerts.png)
```
Comment (TwiN) [inline README.md:737]:
```suggestion
![Gitea alert](.github/assets/gitea-alerts.png)
```

**Review by TwiN — CHANGES_REQUESTED** [review]

File: alerting/provider/gitea/gitea.go (hunk 1 above)
Comment (appleboy) [reply alerting/provider/gitea/gitea.go:24]: Sounds Good. @TwiN

Comment (appleboy) [reply alerting/provider/gitea/gitea.go]: See https://github.com/TwiN/gatus/pull/842/commits/f456cf82c49cf5c34003ed9e0e695eaa0dfa2910

File: README.md [hunk 3]
```diff
@@ -697,6 +699,43 @@ endpoints:
 
 ![GitHub alert](.github/assets/github-alerts.png)
 
+#### Configuring Gitea alerts
+
```
[the remark this answers is not in the document]
Comment (appleboy) [reply README.md]: see [e24209b](https://github.com/TwiN/gatus/pull/842/commits/e24209b722425ef3116a7dd69faa9cfec1030136)

**Comment by appleboy** [comment]
@TwiN All Done. Thanks for your review.

**Review by TwiN — APPROVED** [review]
Excellent work!

**Pull request merged**